docs/extensibility/extend-rest-api-update-cart.md

```diff
@@ -0,0 +1,208 @@
+    /* Do some processing here */
+}
+function remove_discount() {
+    /* Do some processing here */
+}
+
+add_action('woocommerce_blocks_loaded', function() {
+ // ExtendRestApi is stored in the container as a shared instance between the API and consumers.
+ // You shouldn't initiate your own ExtendRestApi instance using `new ExtendRestApi` but should always use the shared instance from the Package dependency injection container.
+ $extend = Package::container()->get( ExtendRestApi::class );
+ $extend->register_update_callback(
+   [
+    'namespace' => 'extension-unique-namespace',
+    'callback'  => function( $data ) {
+      if ( $data['action'] === 'add' ) {
+          add_discount( );
+      }
+      if ( $data['action'] === 'remove' ) {
+          remove_discount();
+      }
+    }
+   ]
+ );
+} );
+```
+If you try to register again, under the same namespace, the previously registered callback will be overwritten.
+
+## API Definition
+
+### PHP
+`ExtendRestApi::register_update_callback`: Used to register a callback to be executed when the `cart/extensions`
+endpoint gets hit with a given namespace. It takes an array of arguments
+
+| Attribute | Type | Required | Description |
+|---|---|---|---|
+| `namespace` | `string` | Yes | The namespace of your extension. This is used to determine which extension's callbacks should be executed. | 
+| `callback` | `Callable` | Yes | The function/method (or Callable) that will be executed when the `cart/extensions` endpoint is hit with a `namespace` that matches the one supplied. The callable should take a single argument. The data passed into the callback via this argument will be an array containing whatever data you choose to pass to it. The callable does not need to return anything, if it does, then its return value will not be used.
+
+### JavaScript
+
+`extensionCartUpdate`: Used to signal that you want your registered callback to be executed, and to pass data to the callback. It takes an object as its only argument.
+
+| Attribute | Type | Required | Description |
+|---|---|---|---|
+| `namespace` | `string` | Yes | The namespace of your extension. This is used to determine which extension's callbacks should be executed. | 
+| `data` | `Object` | No | The data you want to pass to your callback. Anything in the `data` key will be passed as the first (and only) argument to your callback as an associative array.
+
+## Putting it all together
+You are the author of an extension that lets the shopper redeem points that they earn on your website for a discount on
+their order. There is a text field where the shopper can enter how many points they want to redeem, and a submit button 
+that will apply the redemption.
+
+Your extension adds these UI elements to the sidebar in the Cart and Checkout blocks using the [`DiscountsMeta`](./available-slot-fills.md) Slot.
+
+More information on how to use Slots is available in our [Slots and Fills documentation](./slot-fills.md), and thus we
```

Comment:
**nitpick** Instead of

> More information on how to use Slots is available in our [Slots and Fills documentation](./slot-fills.md), and thus we will not be demonstrating how to use them in _this_ document.

I'd write

> More information on how to use Slots is available in our [Slots and Fills documentation](./slot-fills.md).

as adding the second part feels a bit passive-aggressive to me.